frontend/src/metabase-lib/compat/utils.ts

```diff
@@ -0,0 +1,75 @@
+import _ from "underscore";
+import LegacyFilter from "../queries/structured/Filter";
+import { externalOp } from "../common";
+import { filterableColumns, filterableColumnOperators } from "../filter";
+import { displayInfo } from "../metadata";
+import type {
+  ColumnMetadata,
+  ColumnWithOperators,
+  FilterClause,
+  Query,
+} from "../types";
+
+// We'll need to pass it everywhere, just hardcoding to last stage for simplicity
+const stageIndex = -1;
+
+function findFilterableColumn(query: Query, column: ColumnMetadata) {
+  const columnName = displayInfo(query, stageIndex, column).longDisplayName;
+  return filterableColumns(query, stageIndex).find(col => {
+    const info = displayInfo(query, stageIndex, col);
+    return info.longDisplayName === columnName;
```

Comment:
We won't actually need this function (as most of the stuff in `compat/utils`) if `externalOp` returns a column as `ColumnWithOperators` instead of `ColumnMetadata`

We shouldn't do these checks, but comparing `longDisplayName` is the most robust way to compare columns so far (fortunately we didn't need that before, haha). Comparing field IDs won't work because not every column has an ID (e.g. aggregated or custom columns). OTOH, `longDisplayName` has a table name for foreign columns, so the risk of having two identical names is low. We should avoid using it for real though